plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -124,8 +127,27 @@ public void output(Object testId, TestOutputEvent testOutputEvent) {
         delegate.output(testId, testOutputEvent);
     }
 
-    @Override
+    @SuppressWarnings("unused")
     public void failure(Object testId, Throwable throwable) {
+        // Gradle 7.6 changed the method signature from failure(Object, Throwable) to failure(Object, TestFailure).
+        // To maintain compatibility with older versions, the original method needs to exist and needs to call failure()
+        // on the delegate via reflection.
+        failure(testId);
+        try {
+            Method failureMethod = delegate.getClass().getMethod("failure", Object.class, Throwable.class);
```

Comment:
❓ Should we cache the resulting method instead of performing this lookup each time?

Reply:
❌ To answer my own question: I think we should. For simplicity we can look it up in the constructor and cache it as an instance variable.

Reply:
Sorry for not responding earlier; will do that.

Reply:
The method will method only exists for Gradle versions <7.6, so we cannot do the lookup in the constructor (unless can investigate the Gradle version in this class, but that seems like an overkill). A mutable variable cache should suffice.

Reply:
Done